opal/runtime/opal_progress.c

```diff
@@ -50,6 +50,38 @@ bool opal_progress_debug = false;
 static int opal_progress_event_flag = OPAL_EVLOOP_ONCE | OPAL_EVLOOP_NONBLOCK;
 int opal_progress_spin_count = 10000;
 
+/* track whether the async progress thread was spawned */
+/* default=disabled. Enabled with runtime env var OPAL_ASYNC_PROGRESS=1 */
+bool opal_async_progress_thread_spawned = false;
+
+#if OPAL_ENABLE_PROGRESS_THREADS == 1
+/* async progress thread info & args */
+typedef struct thread_args_s {
+    /* number of events processed.
+     * This is updated by the async progress thread and only used for
+     * internal tracking purpose */
+    opal_atomic_int64_t nb_events_processed;
```

Comment:
I don't understand why do you need two atomic variables for the bookkeeping. According to your comment, you expect one to be set by the progress thread and read by the others, and one set by the others. Thus, only one needs to be atomic, the `nb_events_reported`.

Looking at the code it seems you can do everything with a single atomic variable, atomically updated by all threads. The progress thread will add the number of events, while the others will compare_and_swap with 0.

Reply:
You're right, `nb_events_reported` suffices between the two threads. I've added `nb_events_processed` thinking that we may want to keep track of the total number of processed events, since `nb_events_reported` is read-and-clear-and-forget by the main thread. But this may be useless as you say. I'll remove it.